### PR TITLE
imgcodecs cmake: the option to customize supported formats list

### PR DESCRIPTION
**Pull request by Turim**
(WITH_HDR, WITH_SUNRASTER, WITH_PXM)

There were (and may be still exist) vulnerabilities within decoders.
The proposal is "if JPEG is could be excluded let's provide an option to exclude even more specific image formats."

### CODE REVIEW

**Comment by alalek** [comment]
Thank you for the contribution!

This patch should [go](https://github.com/opencv/opencv/wiki/How_to_contribute#faq) into 3.4 branch. So, please:
- change "base" branch of this PR: master => 3.4 (use "Edit" button near PR title)
- rebase your commits from master onto 3.4 branch. For example:
    `git rebase -i --onto upstream/3.4 upstream/master`
    (check list of your commits, save and quit (Esc + "wq" + Enter)
    where `upstream` is configured by following [this GitHub guide](https://help.github.com/articles/configuring-a-remote-for-a-fork/) and fetched (`git fetch upstream`).

We will merge changes from 3.4 into master regularly (weekly/bi-weekly).

File: cmake/OpenCVFindLibsGrfmt.cmake [hunk 1]
```diff
@@ -218,14 +218,6 @@ if(WITH_GDAL)
     endif()
 endif()
 
-if (WITH_GDCM)
-  find_package(GDCM QUIET)
```
Comment (alalek) [inline cmake/OpenCVFindLibsGrfmt.cmake]: This block is still required.

**Comment by vpisarev** [comment]
I think, at some point we need to normalize our CMake flags, 'cause now it's a real mess. As a part of this process, I suggest to use `WITH_IMGCODEC_*` notation, like `WITH_IMGCODEC_TIFF`

**Comment by alalek** [comment]
@vpisarev This will be done as a separate patch.

File: cmake/OpenCVFindLibsGrfmt.cmake [hunk 2]
```diff
@@ -218,14 +218,6 @@ if(WITH_GDAL)
...
-    ocv_clear_vars(GDCM_VERSION GDCM_LIBRARIES)
-  else()
-    set(HAVE_GDCM YES)
-    # include(${GDCM_USE_FILE})
-    set(GDCM_LIBRARIES gdcmMSFF) # GDCM does not set this variable for some reason
-  endif()
-endif()
+set(HAVE_HDR		WITH_HDR)
+set(HAVE_SUNRASTER	WITH_SUNRASTER)
+set(HAVE_PXM		WITH_PXM)
```
Comment (alalek) [inline cmake/OpenCVFindLibsGrfmt.cmake]: Modern (5+ years) CMake doesn't expand `WITH_` variables, so these `HAVE_` vars are always true.

File: modules/imgcodecs/src/loadsave.cpp [hunk 3]
```diff
@@ -131,8 +131,10 @@ struct ImageCodecInitializer
         decoders.push_back( makePtr<BmpDecoder>() );
         encoders.push_back( makePtr<BmpEncoder>() );
 
+	#ifdef HAVE_HDR
```
Comment (alalek) [inline modules/imgcodecs/src/loadsave.cpp]: No tabs please

Comment (Turim) [reply modules/imgcodecs/src/loadsave.cpp]: I beg your pardon. (from what I see - was fixed already, thanks!)

**Pull request merged**